### PR TITLE
add a binding validation unit test

### PR DESCRIPTION
**Pull request by jianzhangbjz**
I just want to use this Travis CI to test my e2e case. And, one question described in below, @pmorie Could you help to clarify it?

### CODE REVIEW

File: pkg/apis/servicecatalog/validation/binding_test.go [hunk 1]
```diff
@@ -564,6 +564,15 @@ func TestValidateServiceBinding(t *testing.T) {
...
 			valid: true,
 		},
+		{
+			name: "empty unbind status on update",
+			binding: func() *servicecatalog.ServiceBinding {
+				b := validServiceBinding()
+				b.Status.UnbindStatus = ""
+				return b
+			}(),
+			valid: false,
```
Comment (jianzhangbjz) [inline pkg/apis/servicecatalog/validation/binding_test.go]: Failed when set the `valid: true`, see below, but as @pmorie mentiond in [bug 1566328](https://bugzilla.redhat.com/show_bug.cgi?id=1566328#c5), there should be a conversion that sets the field value to the right value (NotRequired) if it is unset. So, how to add this conversion in e2e test? I could NOT find the conversion, maybe I misunderstand something. 
```go
[root@localhost validation]# go test -v -test.run TestValidateServiceBinding
=== RUN   TestValidateServiceBinding
--- PASS: TestValidateServiceBinding (0.00s)
PASS
ok  	github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/validation	0.008s
[root@localhost validation]# vim binding_test.go 
[root@localhost validation]# go test -v -test.run TestValidateServiceBinding
=== RUN   TestValidateServiceBinding
--- FAIL: TestValidateServiceBinding (0.00s)
	binding_test.go:582: empty unbind status on update: unexpected error: [status.unbindStatus: Unsupported value: "": supported values: "Required", "Succeeded", "Failed", "NotRequired"]
FAIL
exit status 1
FAIL	github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/validation	0.007s
```

Comment (MHBauer) [reply pkg/apis/servicecatalog/validation/binding_test.go]: Not sure where the conversion could be done, but if it's not done by validation, the next best place would be an integration test. From there you can work backwards to see if it gets set somewhere that is testable by an integration test.

Comment (MHBauer) [reply pkg/apis/servicecatalog/validation/binding_test.go]: looks like it should hit something in this block
https://github.com/jianzhangbjz/service-catalog/blob/1e72d2cde11bfea2ccbf42a53ca52daee87f7ea9/pkg/apis/servicecatalog/validation/binding.go#L151-L159

Comment (MHBauer) [reply pkg/apis/servicecatalog/validation/binding_test.go]: Yes, you have the right output. Working as designed here, maybe a good test to add.

**Comment by MHBauer** [comment]
This is a unit test, not an integration or e2e test.

**Comment by jianzhangbjz** [comment]
@MHBauer Ok, I see, thank you for your clarifying.

File: pkg/apis/servicecatalog/validation/binding_test.go (hunk 1 above)
Comment (MHBauer) [reply pkg/apis/servicecatalog/validation/binding_test.go]: https://github.com/kubernetes-incubator/service-catalog/issues/2185 created as explanation / follow up.

**Comment by jboyd01** [comment]
This unit test looks fine to me to add as it is, but I don't think its accomplishing what @jianzhangbjz wanted.  I believe he wanted a test that verifies Catalog changes an empty UnbindStatus to NotRequired which to my knowledge it does not do.  Re-reading Paul's entry from [bug 1566328](https://bugzilla.redhat.com/show_bug.cgi?id=1566328#c5), 
"- Add a conversion that sets the field value to the right value (NotRequired) if it is unset"

@jianzhangbjz I think I'd change the approach here:  make the test validate what you want (line 574, set valid: **true**) and then as @MHBauer pointed to, add code perhaps in this [block](https://github.com/jianzhangbjz/service-catalog/blob/1e72d2cde11bfea2ccbf42a53ca52daee87f7ea9/pkg/apis/servicecatalog/validation/binding.go#L151-L159) that sets the value to *NotRequired* if the Status.UnbindStatus is not set.

**Comment by MHBauer** [comment]
Validation is not the appropriate place to set fields.

**Comment by jboyd01** [comment]
After discussing with pmorie I'm going to close this issue:

1)  This unit test is testing creation of a Binding with a nil Status.UnbindStatus.  The validators reject this, indicating you must set the UnbindStatus to a valid value.  This is correct behavior - it shouldn't be changed.

2)  When Paul originally wrote up the BZ issue it was dealing with migrating Catalog data from one version to a new version.  The data was being read from old n-1 version format and re-persisted in new N version format.  It was in this process he was thinking we might default the value if not set.  However, he retracts that comment and that is why the original issue and PR didn't deal with it.  The value can not be defaulted and the the Binding Controller will determine what to do with it.

 @jianzhangbjz thanks for the PR anyway, let me know if you want to discuss

**Comment by jianzhangbjz** [comment]
@jboyd01 Sorry for the late reply. Got it, thanks!

**Pull request closed without merging**